19_filter_items/app.js

```diff
@@ -0,0 +1,122 @@
+const buttons = document.querySelectorAll(".button");
+const items = document.querySelectorAll(".card");
+
+// Filter
+
+buttons.forEach(function (button) {
+  button.addEventListener("click", function (e) {
+    console.log(e);
+
+    e.preventDefault;
+    const filter = e.target.dataset.filter;
+    console.log(filter);
+    items.forEach(function (item) {
+      if (item.classList.contains(filter)) {
+        item.style.display = "block";
+      } else {
+        item.style.display = "none";
+      }
+    });
+  });
+});
+
+//Search-bar
+
+const input = document.querySelector(".search-bar");
+const names = document.querySelectorAll(".item-name");
+
+input.addEventListener("keyup", (e) => {
+  //input field sring
+  const filter = e.target.value.toLowerCase().trim();
+  //applying filter to search
+  names.forEach(function (name) {
+    if (name.textContent.toLowerCase().includes(filter)) {
+      name.parentElement.parentElement.style.display = "block";
```

Comment:
Another approach (easier in my opinion)
- Loop through all the cards instead of names.
- Find the .item-name inside the card and check if the text it's a match.

Advantages:
- don't really so heavily on the DOM structure. `parentElement.parentElement` is "sensible".
- you can compare more things (eg. compare name and compare a brand (if there was any)